doc/api/documentation.md

```diff
@@ -40,9 +40,10 @@ The stability indices are as follows:
 
 <!-- separator -->
 
-> Stability: 3 - Legacy. The feature is no longer recommended for use. While it
-> likely will not be removed, and is still covered by semantic-versioning
-> guarantees, use of the feature should be avoided.
+> Stability: 3 - Legacy. It is the same as a forever Documentation-only
+> deprecation. While it likely will not be removed, and is still
+> covered by semantic-versioning guarantees, use of the feature is no longer
+> recommended as it is not officially supported by the project.
```

Comment:
I think likewise the added context here isn't helpful to users. The original text tells them what they need to know clearly. The added context raises more questions than it answers.

Reply:
Moved the relevant part of the text to `deprecations.md`. Does this look better?